Makefile

```diff
@@ -29,15 +29,13 @@ carthage:
 	./carthage.sh build --no-skip-current --use-xcframeworks;
 
 spm:
-# For now just check whether we can assemble it
-# TODO: replace it with "swift test --enable-test-discovery --sanitize=thread" when swiftPM resource-related bug would be fixed.
-# https://bugs.swift.org/browse/SR-13560
-	swift build
+	# FIXME: After resolving data races, change the below line to: swift test --sanitize thread
+	swift test
```

Comment:
I was testing the project locally with Thread Sanitizer enabled and it reported one issue (which fails the SPM CI test):

```
WARNING: ThreadSanitizer: data race (pid=76335)
  Write of size 1 at 0x0001057e13ea by thread T7 (mutexes:
write M0):
#0 -[PINDiskCache initializeDiskProperties] PINDiskCache.m:594 (PINCachePackageTests:arm64+0xe920)
#1 __134-[PINDiskCache initWithName:prefix:rootPath:serializer:deserializer:keyEncoder:keyDecoder:operationQueue:ttlCache:byteLimit:ageLimit:]_block_invoke PINDiskCache.m:284 (PINCachePackageTests:arm64+0xa998)
#2 __tsan::invoke_and_release_block(void*) <null>:4312708 (libclang_rt.tsan_osx_dynamic.dylib:arm64e+0x802e4)
#3 _dispatch_client_callout <null>:4312708 (libdispatch.dylib:arm64e+0x43fc)
Previous read of size 1 at 0x0001057e13ea by main thread:
#0 -[PINDiskCache diskStateKnown] PINDiskCache.m:84 (PINCachePackageTests:arm64+0x1b7e4)
#1 -[PINCacheTests testDiskReadingAfterCacheInit] PINCacheTests.m:863 (PINCachePackageTests:arm64+0x42230)
#2 __invoking___ <null>:4312708 (CoreFoundation:arm64e+0x617f0)
Location is heap block of size 320 at 0x0001057e13c0 allocated by main thread:
#0 calloc <null>:4312708 (libclang_rt.tsan_osx_dynamic.dylib:arm64e+0x562cc)
#1 _objc_rootAllocWithZone <null>:4312708 (libobjc.A.dylib:arm64e+0xe740)
#2 __invoking___ <null>:4312708 (CoreFoundation:arm64e+0x617f0)
Mutex M0 (0x0001057e14c0) created at:
#0 pthread_mutex_init <null>:4312708 (libclang_rt.tsan_osx_dynamic.dylib:arm64e+0x31470)
#1 -[PINDiskCache initWithName:prefix:rootPath:serializer:deserializer:keyEncoder:keyDecoder:operationQueue:ttlCache:byteLimit:ageLimit:] PINDiskCache.m:221 (PINCachePackageTests:arm64+0x9eec)
#2 -[PINDiskCache initWithName:prefix:rootPath:serializer:deserializer:keyEncoder:keyDecoder:operationQueue:ttlCache:] PINDiskCache.m:186 (PINCachePackageTests:arm64+0x9950)
#3 -[PINDiskCache initWithName:prefix:rootPath:serializer:deserializer:keyEncoder:keyDecoder:operationQueue:] PINDiskCache.m:166 (PINCachePackageTests:arm64+0x963c)
#4 -[PINDiskCache initWithName:rootPath:serializer:deserializer:operationQueue:] PINDiskCache.m:147 (PINCachePackageTests:arm64+0x9384)
#5 -[PINDiskCache initWithName:rootPath:serializer:deserializer:] PINDiskCache.m:137 (PINCachePackageTests:arm64+0x9160)
#6 -[PINDiskCache initWithName:rootPath:] PINDiskCache.m:130 (PINCachePackageTests:arm64+0x8f6c)
#7 -[PINDiskCache initWithName:] PINDiskCache.m:125 (PINCachePackageTests:arm64+0x8e18)
#8 -[PINCacheTests testDiskReadingAfterCacheInit] PINCacheTests.m:859 (PINCachePackageTests:arm64+0x42048)
#9 __invoking___ <null>:4312708 (CoreFoundation:arm64e+0x617f0)
Thread T7 (tid=23818133, running) is a GCD worker thread
SUMMARY: ThreadSanitizer: data race PINDiskCache.m:594 in -[PINDiskCache initializeDiskProperties]
```

I suggest keeping this out of scope. It can be addressed as a separate issue that eventually leads to enabling Thread Sanitizer within the Xcode project.

Reply:
I pushed a few more changes around unit tests.

I won't push anything new until the reviewers have had a chance to take a look at the current state of this PR and provide feedback.

Reply:
Agree that keeping this out of scope is what we want to do. 👍